### PR TITLE
Service workers, wether, poll, search

### PR DESCRIPTION
**Pull request by ljubar**
Guys please merge this so Baćko can sync and work on it asap

### CODE REVIEW

File: public/js/scripts/contentCaching.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+
+I am leaving this here so you could reuse some parts of it. Be aware of what you are doing
```
Comment (takeit) [inline public/js/scripts/contentCaching.js:3]: why is it commented if not used?

Comment (ljubar) [inline public/js/scripts/contentCaching.js]: hm not sure.. this is part of changes suggested by Tomek re: content precaching

Comment (ljubar) [reply public/js/scripts/contentCaching.js:3]: That content caching is not entirely finished, still needs to be reviewed by Tomek and then I will remove it if we don't need it, or remove comments from parts that we need

**Comment by takeit** [comment]
I will merge it for now, but next PRs will need to be review properly.

**Pull request merged**